src/db/awsES/locations-dao.js

```diff
@@ -0,0 +1,203 @@
+    }
+  }
+
+  if (parsedParams.evParkingSpaceCount !== undefined) {
+    if (parsedParams.evParkingSpaceCount.operator === '>=') {
+      q.must(esb.rangeQuery('attributes.evParkingSpaceCount')
+        .gte(parsedParams.evParkingSpaceCount.value));
+    }
+  }
+
+  if (parsedParams.bannerAbbreviation !== undefined) {
+    q.must(esb.matchQuery('attributes.bannerAbbreviation', parsedParams.bannerAbbreviation));
+  }
+
+  if (parsedParams.arcGisAbbreviation !== undefined) {
+    q.must(esb.matchQuery('attributes.arcgisAbbreviation', parsedParams.arcGisAbbreviation));
+  }
+
+  // unable to test because nothing is open :(
+  if (parsedParams.isOpen !== undefined) {
+    if (parsedParams.isOpen) {
+      const currentDayIndex = new Date().getDay();
+      q.must(esb.rangeQuery(`attributes.openHours[${currentDayIndex}].start`).lte('now'));
+      q.must(esb.rangeQuery(`attributes.openHours[${currentDayIndex}].end`).gte('now'));
+    } else {
+      q.mustNot(esb.existsQuery('attributes.openHours'));
+    }
+  }
+
+  if (parsedParams.campus !== undefined) {
+    if (parsedParams.campus.operator === 'oneOf') {
+      q.must(esb.termsQuery('attributes.campus', parsedParams.campus.value));
+    }
+  }
+
+  if (parsedParams.parkingZoneGroup !== undefined) {
+    if (parsedParams.parkingZoneGroup.operator === 'oneOf') {
+      q.must(esb.termsQuery('attributes.parkingZoneGroup', parsedParams.parkingZoneGroup.value));
+    }
+  }
+
+  /*
+  geo distance query not working.
+  Maybe has to do with the naming of the lat and lon fields in index?
+
+  if (parsedParams.coordinates !== undefined) {
+    const latitude = parsedParams.coordinates[0];
+    const longitude = parsedParams.coordinates[1];
+    const distance = `${parsedParams.distance}${parsedParams.distanceUnit}`;
+    console.log(parsedParams.coordinates);
+    q.must(esb.geoDistanceQuery()
+      .field('attributes.geoLocation')
+      .distance(distance)
+      .geoPoint(esb.geoPoint().lat(latitude).lon(longitude)));
+  }
+  */
+
+  /* Will implement after GET /locations
+  if (queryParams['include'] !== undefined) {}
+  */
+  return esb.requestBodySearch().query(q).toJSON();
+};
+
+/**
+ * Return a list of Locations
+ * @param queryParams Object containing query parameters
+ * @returns {Promise} Promise object represents a list of locations
+ */
+// GET /locations only returns 10 objects?
+const getLocations = async (queryParams) => {
+  const client = elasticsearch.Client({
+    host: domain,
+    log: 'error',
+    connectionClass,
+    awsConfig: new AWS.Config({
+      accessKeyId,
+      secretAccessKey,
+      region: 'us-east-2',
+    }),
+  });
+
+  console.log(buildQueryBody(queryParams).query);
+  const res = await client.search({
+    index: 'locations',
+    body: buildQueryBody(queryParams),
+  });
+
+  const locations = [];
+  _.forEach(res.hits.hits, (location) => {
+    // eslint-disable-next-line dot-notation
+    const locationSource = location['_source'];
```

Comment:
does `location._source` not work? Worked in browser console.

Reply:
Might be able to do `const { _source: locationSource } = location`

Reply:
using `location._source` throws a 'dangling underscore' error from the linter. Using `location['_source']` also give a linter error saying that dot notation is preferred. Since it errors both ways, I figured the using bracket notation would look better

Reply:
I tested myself and `const { _source: locationSource } = location;` works and you don't need to disable the lint rule.

Reply:
@kalcicd did you address Alex's comment here?